### PR TITLE
chore(github): create dependabot.yml

### PR DESCRIPTION
**Pull request by joshka**
This is the default dependabot yaml file that is generated by github

### CODE REVIEW

File: .github/dependabot.yml
Comment (orhun) [inline .github/dependabot.yml:1]: Should we also check the GitHub Actions dependencies? Like so:

```yml
  # Maintain dependencies for GitHub Actions
  - package-ecosystem: github-actions
    directory: "/"
    schedule:
      interval: weekly
    open-pull-requests-limit: 10

```

Comment (joshka) [reply .github/dependabot.yml:1]: yes

Comment (joshka) [reply .github/dependabot.yml:1]: https://github.com/ratatui-org/ratatui/commit/8a312962a41254d33e26b7b308a6e559b3a96d76

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "cargo" # See documentation for possible values
```
Comment (orhun) [inline .github/dependabot.yml:8]:
```suggestion
  # Maintain dependencies for Cargo
  - package-ecosystem: "cargo" # See documentation for possible values
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  # Maintain dependencies for Cargo
+  - package-ecosystem: "cargo" # See documentation for possible values
```
Comment (joshka) [inline .github/dependabot.yml:9]:
```suggestion
  - package-ecosystem: "cargo"
```

**Review by orhun — APPROVED** [review]

**Pull request merged**